codex/sales/reservations.nim

```diff
@@ -497,12 +513,13 @@ proc storables(
 
 proc all*(
   self: Reservations,
-  T: type SomeStorableObject
+  T: type SomeStorableObject,
+  availabilityId: ?AvailabilityId = AvailabilityId.none
```

Comment:
Perhaps instead we could add an `all` overload that accepts an availabilityId instead of making it optional. That overload would create a filter predicate that can be passed to `storables`, which would insert that filtering predicate (matching an availabilityId) into the `next` query logic.

In this overload (without the availabilityId parameter), we would pass a `none` predicate to storables.